internal/pkg/agent/storage/encrypted_disk_store.go

```diff
@@ -0,0 +1,168 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package storage
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+
+	"github.com/elastic/elastic-agent-libs/file"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/secret"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/crypto"
+	"github.com/hectane/go-acl"
+)
+
+const darwin = "darwin"
+
+var encryptionDisabled bool
+
+// DisableEncryption disables storage encryption.
+// Is needed for existing unit tests on Mac OS, because the system keychain requires sudo
+func DisableEncryptionDarwin() {
+	if runtime.GOOS == darwin {
+		encryptionDisabled = true
+	}
+}
+
+type OptionFunc func(s *EncryptedDiskStore)
+
+// NewEncryptedDiskStore creates an encrypted disk store.
+// Drop-in replacement for NewDiskStorage
+func NewEncryptedDiskStore(target string, opts ...OptionFunc) Storage {
+	if encryptionDisabled {
```

Comment:
Do we want to emit a message of some sort if it returns a normal store?

Reply:
This option should not be used normally. 
This is only used to keep the existing unit tests working that are not ran with sudo